client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogmaBuilder.java

```diff
@@ -29,30 +45,195 @@
 import com.linecorp.centraldogma.internal.CsrfToken;
 import com.linecorp.centraldogma.internal.thrift.CentralDogmaService;
 
+import io.netty.util.NetUtil;
+
 /**
  * Builds a {@link CentralDogma} client.
- *
- * <pre>{@code
- * CentralDogmaBuilder builder = new CentralDogmaBuilder();
- * builder.uri("tbinary+http://example.com:36462/cd/thrift/v1");
- * CentralDogma dogma = builder.build();
- * }</pre>
  */
 public class CentralDogmaBuilder {
 
+    private static final int DEFAULT_PORT = 36462;
+
+    private static final AtomicLong nextGroupId = new AtomicLong();
+
     private ClientFactory clientFactory = ClientFactory.DEFAULT;
-    private URI uri;
+    private final List<Endpoint> endpoints = new ArrayList<>();
     private ArmeriaClientConfigurator clientConfigurator = cb -> {
     };
 
     /**
-     * Sets the {@link URI} of the Central Dogma server.
+     * Adds the {@link URI} of the Central Dogma server.
+     *
+     * @deprecated Use {@link #host(String)} or {@link #profile(String...)} instead.
      *
      * @param uri the URI of the Central Dogma server. e.g.
      *            {@code tbinary+http://example.com:36462/cd/thrift/v1}
      */
+    @Deprecated
     public CentralDogmaBuilder uri(String uri) {
-        this.uri = URI.create(requireNonNull(uri, "uri"));
+        final URI parsed = URI.create(requireNonNull(uri, "uri"));
+        final String host = parsed.getHost();
+        final int port = parsed.getPort();
+        checkArgument(host != null, "uri: %s (must contain a host part)");
```

Comment:
Could you add `host` to arguments to show up `host` to error message?
`checkArgument(host != null, "uri: %s (must contain a host part)", host)`

Reply:
Oops. Added `uri` as the last parameter of `checkArgument()`